### PR TITLE
Data encryption on VFS level

### PR DESCRIPTION
**Pull request by alexelex**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
adding encryption at the virtual file system level using AES_128_CTR algorithm
...

Detailed description / Documentation draft:
to declare an encrypted disk as a wrapper to another disk, add to the configuration
```xml
<storage_configuration>
    <disks>
        <encrypted_disk_name>
            <disk>wrapped_disk_name</disk>
            <type>encrypted</type>
            <key>enter_key</key>
            <path>path_to_encrypted_disk_on_wrapped_disk</path>
        </encrypted_disk_name>
    </disks>
</storage_configuration>
```
Each file will be encrypted using the key and AES-128-CTR algorithm 

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by alexelex** [comment]
There are two issues now:
- it might be better to set `HAVE_SYS_RANDOM_H` (or whatever, to use `sys/random.h` - `getrandom`) somehow differently, because it compiles for me without this flag, but when cmake tries `check_include_file(sys/random.h) it returns 0.
- `GRND_RANDOM` flag is better for generation random IV than 0 or `GRND_NONBLOCK` (based on [stackoverflow](https://stackoverflow.com/questions/23712581/differences-between-random-and-urandom) , [stackexchange](https://unix.stackexchange.com/questions/324209/when-to-use-dev-random-vs-dev-urandom#:~:text=There%20is%20no%20difference%20between,Apple's%20iOS%20also%20uses%20Yarrow.) ), but it works very slow. And I'm not sure which is best for CH.

File: src/Functions/FileEncryption.h [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+            buf.write(EncryptNBytes(plaintext + part_size, size, iv).data(), size);
+            offset += size;
+        }
+    }
+
+private:
+    String EncryptPartialBlock(const char * partial_block, size_t size, const InitVector & iv, size_t off) const
+    {
+        if (size > block_size)
+            throw DB::Exception("Expecter partial block, got block with size > block_size: size = " + std::to_string(size) + " and offset = " + std::to_string(off),
```
Comment (alexey-milovidov) [inline src/Functions/FileEncryption.h:232]: Expecter -> Expected

**Comment by vitlibar** [comment]
@alexelex Very good work, thank you!

**Comment by sevirov** [comment]
Internal documentation ticket: DOCSUP-12128

**Pull request merged**